src/transaction.rs

```diff
@@ -613,10 +615,25 @@ impl Transaction {
         bitcoin::Wtxid::from_engine(enc)
     }
 
-    /// Get the total transaction fee.
-    pub fn fee(&self) -> u64 {
+    /// Get the total transaction fee in the given asset.
+    pub fn fee_in(&self, asset: AssetId) -> u64 {
```

Comment:
maybe simpler calling all_fees() ?

Reply:
I'd like to avoid building the hashmap etc.